### PR TITLE
Support Rails 7.1

### PR DESCRIPTION
**Pull request by mgrunberg**
Ports #8098, #8105, #8106 and #8102 into default branch.

### CODE REVIEW

**Comment by mgrunberg** [comment]
@javierjulio please review this PR

**Review by javierjulio — COMMENTED** [review]
@mgrunberg thank you. Don't we want to include [the flaky test case fix commit](https://github.com/activeadmin/activeadmin/pull/8102/commits/a8544997fd276b893c597a1e16f524f20b7ae4c5) here though? Other than that, everything looks good.

**Comment by mgrunberg** [comment]
> @mgrunberg thank you. Don't we want to include [the flaky test case fix commit](https://github.com/activeadmin/activeadmin/pull/8102/commits/a8544997fd276b893c597a1e16f524f20b7ae4c5) here though? Other than that, everything looks good.

@javierjulio I didn't include it because spec changed and is not mocking/stubbing I18n anymore (there is no  `expect(I18n).to receive(:t).at_least(:once)`). There is no need for that fix here.

File: .eslintignore [hunk 1]
```diff
@@ -6,4 +6,5 @@ spec/**/*.js
 src/**/*.js
 coverage/**/*.js
 rails_70/**/*.js
+rails_71/**/*.js
```
[the remark this answers is not in the document]
Comment (mgrunberg) [reply .eslintignore]: Thanks for the clarification

**Comment by javierjulio** [comment]
@mgrunberg thanks for clarifying. Yes, that makes sense. I made changes there for the TailwindCSS migration so it's no longer applicable on the default branch. I missed that.

**Review by javierjulio — APPROVED** [review]
Thanks!

**Pull request merged**